core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerProxy.scala

```diff
@@ -714,14 +715,19 @@ class ContainerProxy(factory: (TransactionId,
       } else {
         CompletionMessage(tid, result, instance)
       }
-      sendActiveAck(tid, result, job.msg.blocking, job.msg.rootControllerIndex, job.msg.user.namespace.uuid, msg)
-        .andThen {
-          case Failure(e) => logging.error(this, s"failed to send abort ack $e")
-        }
-      storeActivation(tid, result, job.msg.blocking, context).andThen {
-        case Failure(e) => logging.error(this, s"failed to store aborted activation $e")
+      val ack =
+        sendActiveAck(tid, result, job.msg.blocking, job.msg.rootControllerIndex, job.msg.user.namespace.uuid, msg)
+          .recover {
```

Comment:
Do you really want to recover (make the Future succeed) here rather than just log? Same below. The sequence below will always succeed.

Reply:
My concern is when one runs, and fails, not affecting the others. For both ack vs store, as well as processing each item in queue. I can guess though that if one item fails, that the other items will fail as well, so maybe I would just go back to `.andThen` processing to log failures.